docs/madr/decisions/026-meshgateway-targetref.md

```diff
@@ -0,0 +1,122 @@
+`MeshGateway` resource where the listener configuration is held.
+
+It may be more intuitive to allow referring directly to a `MeshGateway` resource
+but this isn't yet supported.
+
+Another decision is how a policy can be restricted to a single listener of a
+`MeshGateway`.
+
+This MADR does not address narrowing to a specific route resource or route match.
+
+### `spec.to.targetRef`
+
+This MADR also makes clear that currently `spec.to.targetRef.kind: MeshGateway`
+to set policy for incoming traffic is not supported in any policy.
+
+Part of the reason for this is that the paradigm that `spec.targetRef` tells us
+which sidecars have their config modified by a given policy would no longer apply.
+
+### Policies targetting incoming traffic
+
+We should consider which policies make sense with `spec.targetRef.kind: MeshGateway`
+for targetting _incoming traffic_, i.e. not using `spec.to` but rather `spec.from`.
+For this kind of policy we can imagine a new `spec.from.targetRef.kind: External`.
+
+## Decision Drivers
+
+- Allow users to refer directly to the `MeshGateway` resource itself
+
+## Considered Options
+
+- Leave out `MeshGateway` as a `kind`
+- Support `MeshGateway` as a `kind` but not specific listeners
+- Support `MeshGateway` as a `kind` with the option of setting
+  `targetRef.listener`
+- Support `MeshGateway` as a `kind` but also `MeshGatewayListener` where
+  `targetRef.listener` can be set
+- Support `MeshGateway` as a `kind` with the option of setting
+  `targetRef.sectionName`
+
+## Decision Outcome
+
+Chosen option: `MeshGateway` as `kind` with `targetRef.sectionName`
+
+`targetRef.kind: MeshGateway` applies the policy to all `Dataplanes` matched by
+the matchers on the `MeshGateway` object and all listeners configured in the
+`MeshGateway`. It has the same semantics as `{ kind: MeshService, name:
+<kuma.io/service matched by the MeshGateway> }`
+
+Further specifying `sectionName`
+narrows this to a specific listener. `MeshGateway` listeners don't have explicit
+names but users can use a reserved tag to give them a canonical name that
+policies can match on. The Gateway API implementation uses the
+`gateways.kuma.io/listener-name` tag, for example, which we could reuse.
+
+Given:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshGateway
+metadata:
+  name: edge-gateway
+mesh: default
+spec:
+  selectors:
+    - match:
+        kuma.io/service: edge-gateway
+  conf:
+    listeners:
+      - port: 8080
+        protocol: HTTP
+        tags:
+          gateways.kuma.io/listener-name: http
+      - port: 8443
+        protocol: HTTPS
+        tags:
+          gateways.kuma.io/listener-name: https
+```
+
+We can use the following policy with `sectionName` to target only traffic over
+the `HTTPS` listener on port `8443`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: SomePolicy
+metadata:
+  name: https-only
+spec:
+  targetRef:
+    kind: MeshGateway
+    name: edge
+    sectionName: https
```

Comment:
It's a good point, I don't have a super strong opinion. [`sectionName` looks like something the Gateway API policy attachment proposal might have in the future](https://gateway-api.sigs.k8s.io/geps/gep-713/#apply-policies-to-sections-of-a-resource-future-extension). So it might be the time to introduce/limit things to that.

Reply:
With `sectionName` you still have to explain in docs that in this particular case `sectionName` means "listener". So in my opinion `sectionName` doesn't make things more user-friendly. 

Another point is that `tags` are more versatile, you can use them to select a group of listeners (not just 1 listener by its name).

In general, I feel like it's better to not introduce a new field, especially in the "core" than introduce it. We can always add a `sectionName` in the future if Gateway API moves forward with it

Reply:
I'm going to change the MADR to using `tags`.